src/contracts/core/ReleaseManager.sol

```diff
@@ -0,0 +1,88 @@
+// SPDX-License-Identifier: BUSL-1.1
+pragma solidity ^0.8.27;
+
+import "@openzeppelin-upgrades/contracts/proxy/utils/Initializable.sol";
+import "@openzeppelin/contracts/utils/Strings.sol";
+import "../mixins/PermissionControllerMixin.sol";
+import "../mixins/SemVerMixin.sol";
+import "./ReleaseManagerStorage.sol";
+
+contract ReleaseManager is Initializable, ReleaseManagerStorage, PermissionControllerMixin, SemVerMixin {
+    using EnumerableSet for EnumerableSet.Bytes32Set;
+    using OperatorSetLib for OperatorSet;
+    using Strings for uint16;
+
+    /**
+     *
+     *                         INITIALIZING FUNCTIONS
+     *
+     */
+    constructor(
+        IAllocationManager _allocationManager,
+        IPermissionController _permissionController,
+        string memory _version
+    )
+        ReleaseManagerStorage(_allocationManager)
+        PermissionControllerMixin(_permissionController)
+        SemVerMixin(_version)
+    {
+        _disableInitializers();
+    }
+
+    /**
+     *
+     *                         EXTERNAL FUNCTIONS
+     *
+     */
+
+    /// @inheritdoc IReleaseManager
+    function publishRelease(
+        OperatorSet calldata operatorSet,
+        Artifact[] calldata artifacts,
+        uint32 upgradeByTime
+    ) external checkCanCall(operatorSet.avs) {
+        Release[] storage releases = _operatorSetReleases[operatorSet.key()];
+
+        uint256 newTotalReleases = releases.length + 1;
+
+        assembly {
+            sstore(releases.slot, newTotalReleases)
+        }
```

Comment:
Thinking out loud here. Is there any benefit to making the releases an `EnumerableSet` instead of an array? I'm assuming we would still need a key, which in your current case is the array Index serves the role.

Reply:
I'm assuming your current assembly is resizing the array right?